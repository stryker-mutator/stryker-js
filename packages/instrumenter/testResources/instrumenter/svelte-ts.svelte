<script lang="ts" context="module">
    function add(a: number, b: number) {
        return a + b;
    }
</script>

<script lang="ts">
    import type { Athlete } from './athlete.js';

    const athletes: Athlete[] = [
      { sport: "Athletics", name: "Alice", age: 16 },
      { sport: "Swimming", name: "Bob", age: 18 },
      { sport: "Tennis", name: "Charlie", age: 20 },
    ];
  
    let a = 0;
    let b = 0;
    $: count = a + b;
</script>

<main>
   <script lang="ts">
    function subtract(a: number, b: number) {
        return a - b;
    }
   </script>
</main>
  