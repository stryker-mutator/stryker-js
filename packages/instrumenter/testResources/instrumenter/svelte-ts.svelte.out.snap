// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instrumenter integration should be able to instrument typescript inside svelte 1`] = `
"<script lang=\\"ts\\" context=\\"module\\">
function stryNS_9fa48() {
  var g = typeof globalThis === 'object' && globalThis && globalThis.Math === Math && globalThis || new Function(\\"return this\\")();
  var ns = g.__stryker__ || (g.__stryker__ = {});
  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }
  function retrieveNS() {
    return ns;
  }
  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}
stryNS_9fa48();
function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });
  function cover() {
    var c = cov.static;
    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }
    var a = arguments;
    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }
  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}
function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();
  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }
      return true;
    }
    return false;
  }
  stryMutAct_9fa48 = isActive;
  return isActive(id);
}
function add(a: number, b: number) {
  if (stryMutAct_9fa48(\\"0\\")) {
    {}
  } else {
    stryCov_9fa48(\\"0\\");
    return stryMutAct_9fa48(\\"1\\") ? a - b : (stryCov_9fa48(\\"1\\"), a + b);
  }
}
</script>

<script lang=\\"ts\\">
import type { Athlete } from './athlete.js';
const athletes: Athlete[] = stryMutAct_9fa48(\\"2\\") ? [] : (stryCov_9fa48(\\"2\\"), [stryMutAct_9fa48(\\"3\\") ? {} : (stryCov_9fa48(\\"3\\"), {
  sport: stryMutAct_9fa48(\\"4\\") ? \\"\\" : (stryCov_9fa48(\\"4\\"), \\"Athletics\\"),
  name: stryMutAct_9fa48(\\"5\\") ? \\"\\" : (stryCov_9fa48(\\"5\\"), \\"Alice\\"),
  age: 16
}), stryMutAct_9fa48(\\"6\\") ? {} : (stryCov_9fa48(\\"6\\"), {
  sport: stryMutAct_9fa48(\\"7\\") ? \\"\\" : (stryCov_9fa48(\\"7\\"), \\"Swimming\\"),
  name: stryMutAct_9fa48(\\"8\\") ? \\"\\" : (stryCov_9fa48(\\"8\\"), \\"Bob\\"),
  age: 18
}), stryMutAct_9fa48(\\"9\\") ? {} : (stryCov_9fa48(\\"9\\"), {
  sport: stryMutAct_9fa48(\\"10\\") ? \\"\\" : (stryCov_9fa48(\\"10\\"), \\"Tennis\\"),
  name: stryMutAct_9fa48(\\"11\\") ? \\"\\" : (stryCov_9fa48(\\"11\\"), \\"Charlie\\"),
  age: 20
})]);
let a = 0;
let b = 0;
$: count = stryMutAct_9fa48(\\"12\\") ? a - b : (stryCov_9fa48(\\"12\\"), a + b);
</script>

<main>
   <script lang=\\"ts\\">
function subtract(a: number, b: number) {
  if (stryMutAct_9fa48(\\"13\\")) {
    {}
  } else {
    stryCov_9fa48(\\"13\\");
    return stryMutAct_9fa48(\\"14\\") ? a + b : (stryCov_9fa48(\\"14\\"), a - b);
  }
}
</script>
</main>
  "
`;
